app/src/controllers/plugins/uiExtensions/sagas.js

```diff
@@ -2,18 +2,22 @@ import { select, call, all, put } from 'redux-saga/effects';
 import { URLS } from 'common/urls';
 import { fetch } from 'common/utils/fetch';
 import { activeProjectSelector } from 'controllers/user';
+import { PUBLIC_PLUGINS } from 'controllers/plugins/constants';
 import { COMMAND_GET_FILE, METADATA_FILE_KEY, MAIN_FILE_KEY } from './constants';
-import { pluginsSelector, globalIntegrationsSelector } from '../selectors';
+import { pluginsSelector, globalIntegrationsSelector, publicPluginsSelector } from '../selectors';
 import { filterIntegrationsByName, isPluginSupportsCommonCommand } from '../utils';
 import {
   extensionLoadFinishAction,
   extensionLoadStartAction,
   fetchExtensionsMetadataSuccessAction,
 } from './actions';
 
-function* fetchExtensionsMetadata() {
+export function* fetchExtensionsMetadata(...params) {
   const plugins = yield select(pluginsSelector);
-  const uiExtensionPlugins = plugins.filter(
+  const publicPlugins = yield select(publicPluginsSelector);
+  const currentPlugins =
+    params[0] && params[0].meta.namespace === PUBLIC_PLUGINS ? publicPlugins : plugins;
+  const uiExtensionPlugins = currentPlugins.filter(
```

Comment:
Please create a constant `isPublic` or smth like that to hold plugin type and use it to select necessary plugins.
Also use this value to choose right URL for metadata fetching in line 37 (pluginFile vs pluginPublicFile urls)